src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11685,6 +11694,14 @@ var tk
 
                 case ScanTypeFlags.GenericTypeOrExpression:
                 case ScanTypeFlags.NonGenericTypeOrExpression:
+                    // if we have `(A)[]` then treat that always as a cast of an empty collection literal.  `[]` is not
```

Comment:
>if we have `(A)[]` then treat that always as a cast of an empty collection literal. [](http://example.com/codeflow?start=23&length=83)

Hmm, it seems surprising to treat `(A)[]` as a collection literal but `(A)[b]` as an indexer. I realize we're already parsing `(A)[b]` and `(A[])[b]` differently but it might be useful to have consistent interpretation of `[...]` for a given form within `(...)`.

Reply:
well, `(A)[]` is *non-ambiguous*, so it's correct to parse as a cast.  However, `(A)[b]` is truly ambiguous and so needs to go through teh LDM discussion around what the right thing is to do with it (continue parsing as an indexer, or take a breaking change and parse as a cast).

Put another way, `(A)[]` was an *error* today.  So it's correct to move it to correctly parse given the new grammar rules we have.